tool/tsh/common/tsh.go

```diff
@@ -705,6 +706,38 @@ func initLogger(cf *CLIConf) {
 //
 // DO NOT RUN TESTS that call Run() in parallel (unless you taken precautions).
 func Run(ctx context.Context, args []string, opts ...CliOption) error {
+	// At process startup, check if a version has already been downloaded to
+	// $TELEPORT_HOME/bin or if the user has set the TELEPORT_TOOLS_VERSION
+	// environment variable. If so, re-exec that version of {tsh, tctl}.
+	toolsDir, err := tools.Dir()
```

Comment:
Also, the RFD says that [Connect is going to call tsh with `TELEPORT_TOOLS_VERSION=off`](https://github.com/gravitational/teleport/blob/c906170e1dc2c7f32a4902c4ba81db2e29efb654/rfd/0144-client-tools-updates.md#out-of-scope), but from what I see this hasn't been implemented – grepping for `TELEPORT_TOOLS_VERSION` doesn't bring up anything.

Reply:
Addressed by https://github.com/gravitational/teleport/pull/49180